CRM/Mailing/BAO/MailingJob.php

```diff
@@ -478,6 +478,40 @@ public function queue($testParams = NULL) {
   }
 
   /**
+   * Build a join and where part for a query
+   *
+   * @param $contact_id
+   * @return array - the first key is join part of the query and the second key is the where part of the query
+   */
+  public function buildAcl($contact_id) {
```

Comment:
Perhaps you could put it on the ACL_BAO class & call it statically? This doesn't seem very mailing specific. 

In the reports classes I see

  /**
   * Build acl clauses.
   *
   * @param string $tableAlias
   */
  public function buildACLClause($tableAlias = 'contact_a') {
    list($this->_aclFrom, $this->_aclWhere) = CRM_Contact_BAO_Contact_Permission::cacheClause($tableAlias);
  }

& I think there is also something on Contact_BAO_Query object